Userland/Libraries/LibGfx/Painter.cpp

```diff
@@ -2443,7 +2443,7 @@ void Painter::draw_text_run(FloatPoint baseline_start, Utf8View const& string, F
 
 void Painter::draw_scaled_bitmap_with_transform(IntRect const& dst_rect, Bitmap const& bitmap, FloatRect const& src_rect, AffineTransform const& transform, float opacity, ScalingMode scaling_mode)
 {
-    if (transform.is_identity_or_translation_or_scale()) {
+    if (transform.is_identity_or_translation_or_scale() && transform.a() > 0 && transform.d() > 0) {
```

Comment:
I wonder if `is_identity_or_translation_or_scale` should take an enum like `AllowsNegativeScale/AllowsFlip::Yes/No`? I suspect (though I've not checked) there could be other places where this is an issue.

Reply:
There's only one other caller and I think it's fine with the old semantics. But it's a good suggestion, so done, thanks!